lnrpc/rpc.proto

```diff
@@ -1742,6 +1742,13 @@ message QueryRoutesRequest {
     A list of directed node pairs that will be ignored during path finding.
     */
     repeated NodePair ignored_pairs = 10;
+
+    /** 
+    An optional maximum total time lock for the route. If the source is empty or
+    ourselves, this should not exceed lnd's `--max-cltv-expiry` setting. If
+    zero, then the value of `--max-cltv-expiry` is used as the limit.
+    */
+    uint32 cltv_limit = 11;
```

Comment:
This is actually required for the base bug fix? I can see how it's a nice to have to ensure users don't obtain a route than we can't send to however.

Reply:
On the other hand, nice that it unifies some code around the handling of the final CLTV delta.

Reply:
We could have left it out and always queryroutes with the max cltv that is configured (if routing from self). I don't mind it being added in the pr.

Reply:
Yep, fine with me as well.